Add GPU benchmarks support to readme (#59)
IntelPython/scikit-learn_bench: README.md

> runner options:

Reviewer: ```suggestion
Options:
```

> * ``configs`` : configuration files paths

Reviewer: ```suggestion
* ``--configs``: specify the path to a configuration file.
```

> * ``no-intel-optimized`` : using Scikit-learn without [Intel(R) Extension for Scikit-learn*](#Intel(R)-Extension-for-Scikit-learn*-support). Now avalible for scikit-learn benchmarks. Default starts with using Intel(R) Extension for Scikit-learn*.

Reviewer 2: Not works reference. Maybe?
```suggestion
* ``no-intel-optimized`` : using Scikit-learn without [Intel(R) Extension for Scikit-learn*](## Intel(R)-Extension-for-Scikit-learn*-support). Now avalible for scikit-learn benchmarks. Default starts with using Intel(R) Extension for Scikit-learn*.
```

> * ``no-intel-optimized`` : using Scikit-learn without [Intel(R) Extension for Scikit-learn*](#Intel(R)-Extension-for-Scikit-learn*-support). Now avalible for scikit-learn benchmarks. Default starts with using Intel(R) Extension for Scikit-learn*.

Reviewer 2: ```suggestion
* ``no-intel-optimized`` : using Scikit-learn without [Intel(R) Extension for Scikit-learn*](#Intel(R)-Extension-for-Scikit-learn*-support). Now available for [scikit-learn benchmarks](https://github.com/IntelPython/scikit-learn_bench/tree/master/sklearn_bench). Default running with using Intel(R) Extension for Scikit-learn.
```

> * ``no-intel-optimized`` : using Scikit-learn without [Intel(R) Extension for Scikit-learn*](#intelr-extension-for-scikit-learn-support). Now available for [scikit-learn benchmarks](https://github.com/IntelPython/scikit-learn_bench/tree/master/sklearn_bench). Default running with using Intel(R) Extension for Scikit-learn.

Reviewer: `Now available for [scikit-learn benchmarks]` -- why do you need this info?

```suggestion
* ``--no-intel-optimized``: use Scikit-learn without [Intel(R) Extension for Scikit-learn*](#intelr-extension-for-scikit-learn-support). Now available for [scikit-learn benchmarks](https://github.com/IntelPython/scikit-learn_bench/tree/master/sklearn_bench). By default, the runner uses Intel(R) Extension for Scikit-learn.
```

> * ``output-file``: output file name for result benchmarks. Default is `result.json`

Reviewer: ```suggestion
* ``--output-file``: output file name for the benchmark result. The default name is `result.json`.
```

> * ``report``: create an Excel report based on benchmarks results. Need library `openpyxl`.

Reviewer: ```suggestion
* ``--report``: create an Excel report based on benchmark results. The `openpyxl` library is required.
```

> * ``dummy-run`` : run configuration parser and datasets generation without benchmarks running.

Reviewer: ```suggestion
* ``--dummy-run``: run configuration parser and dataset generation without benchmarks running.
```

## Benchmark supported algorithms

> | algorithm  | benchmark name | sklearn | sklearn on GPU | daal4py | cuml | xgboost |

Reviewer 2: My concerns:
- What is sklearn on GPU?
- It is not aligned with the rest of the benchmarks for libraries: sklearn/xgboost/daal4py/cuml

if you want to highlight algorithms on GPU, I would make it as a note at the bottom of the table.

@SmirnovEgorRu , @michael-smirnov  What do you think?

|**[GradientBoostingClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingClassifier.html)**|gbt|:x:|:x:|:x:|:white_check_mark:|
|**[GradientBoostingRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingRegressor.html)**|gbt|:x:|:x:|:x:|:white_check_mark:|

## Intel(R) Extension for Scikit-learn* support

> [Intel(R) Extension for Scikit-learn](https://github.com/intel/scikit-learn-intelex) providing drop-in patching speeds up scikit-learn on CPU and GPU. 

Reviewer 3: "providing drop-in patching": there is no final decision yet about how XPUs will be supported. Can we use more neutral wordings like "provides optimized functionality of scikit-learn on Intel(R) CPU and GPU" at the moment?

> Scikit-learn benchmarks on GPU with Intel(R) Extension for Scikit-learn support algorithms: **DBSCAN**, **KMeans**, **LinearRegression**, **LogisticRegression**. Example config with GPU support [here](https://github.com/IntelPython/scikit-learn_bench/blob/master/configs/skl_xpu_config.json).

Reviewer 3: The following benchmarks support GPU computations with help of Intel(R) Extension for Scikit-learn*:
- dbscan
- kmeans ...

Reviewer 3: Example config with GPU support [here]: The configuration file that contains all of these benchmarks can be found [here].
The link for "here" is broken.
Please add an example of command how to run this config on GPU here

|**[GradientBoostingClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingClassifier.html)**|gbt|:x:|:x:|:x:|:white_check_mark:|
|**[GradientBoostingRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingRegressor.html)**|gbt|:x:|:x:|:x:|:white_check_mark:|

## Intel(R) Extension for Scikit-learn support

> By default scikit-learn benchmark launches using [Intel(R) Extension for Scikit-learn](https://github.com/intel/scikit-learn-intelex) on the CPU (use ``no-intel-optimized`` option to run without extention). Some benchmarks have a GPU support:

Reviewer 3: Not sure this is correct sentence. Maybe change by "The launches of scikit-learn benchmark use ... by default"?

> The runs of Scikit-learn benchmark use [Intel(R) Extension for Scikit-learn](https://github.com/intel/scikit-learn-intelex) on the CPU by default (use ``no-intel-optimized`` option to run without extention). Some benchmarks have a GPU support:

Reviewer: ```suggestion
When you run scikit-learn benchmarks on CPU, [Intel(R) Extension for Scikit-learn](https://github.com/intel/scikit-learn-intelex) is used by default. Use the ``--no-intel-optimized`` option to run the benchmarks without the extension. 

The following benchmarks have a GPU support:
```

* dbscan
* kmeans
* linear
* log_reg

> A configuration file that contains all these benchmarks can be found [here](https://github.com/IntelPython/scikit-learn_bench/blob/master/configs/skl_xpu_config.json). You can use this file to run these benchmarks on both CPU and GPU.

Reviewer 3: Please add the command to run this config and get a report

> A configuration file that contains all these benchmarks can be found [here](https://github.com/IntelPython/scikit-learn_bench/blob/master/configs/skl_xpu_config.json). You can use this file to run these benchmarks on both CPU and GPU.

Reviewer: ```suggestion
You may use the [configuration file for these benchmarks](https://github.com/IntelPython/scikit-learn_bench/blob/master/configs/skl_xpu_config.json) to run them on both CPU and GPU.
```

|**[GradientBoostingClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingClassifier.html)**|gbt|:x:|:x:|:x:|:white_check_mark:|
|**[GradientBoostingRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.GradientBoostingRegressor.html)**|gbt|:x:|:x:|:x:|:white_check_mark:|

## Intel(R) Extension for Scikit-learn* support

> [Intel(R) Extension for Scikit-learn](https://github.com/intel/scikit-learn-intelex) provides the ability to run scikit-learn on CPU and GPU with kernels optimized by [oneDAL](https://github.com/oneapi-src/oneDAL). The extension support GPU patching for algorihms: **DBSCAN**, **KMeans**, **LinearRegression**, **LogisticRegression**. You can launch benchmarks with GPU support using [skl_xpu_config.json](https://github.com/IntelPython/scikit-learn_bench/blob/master/configs/skl_with_context_config.json) configuration file.

Reviewer 2: ```suggestion
[Intel(R) Extension for Scikit-learn](https://github.com/intel/scikit-learn-intelex) providing drop-in patching speeds up scikit-learn on CPU and GPU.

Scikit-learn benchmarks on GPU with Intel(R) Extension for Scikit-learn support algorithms: **DBSCAN**, **KMeans**, **LinearRegression**, **LogisticRegression**.  Example config with GPU support [here](https://github.com/IntelPython/scikit-learn_bench/blob/master/configs/skl_with_context_config.json).
```
